supabase.sql

```diff
@@ -51,3 +51,19 @@ $$;
 create trigger on_auth_user_created
   after insert on auth.users
   for each row execute procedure public.handle_new_user();
+
+-- Create a table for questions
+create table questions (
+  id uuid default uuid_generate_v4() primary key,
+  created_at timestamp with time zone default timezone('utc'::text, now()) not null,
+  created_by uuid references auth.users(id) not null,
+  company text,
+  location text,
+  asked_month smallint,
```

Comment:
shouldn't we just put this in a date format? we can use a date type for it, and use the first of the given month

Reply:
We are going to need `asked_month` (0-11) and `asked_year` (4-digit number) to generate the month name and year on the frontend.

```js
new Date(`asked_year`, `asked_month`).toLocaleString('default', { year: 'numeric', month: 'long' });
```
Additionally `smallint` cost 2 bytes whereas `date` costs 4 bytes. We only need to store 0-11 here so `smallint` is more than enough for our use case.

Reply:
I think it's more convenient and intuitive to store a date, though. If we want to do any type of operations in the future (say grouping, plotting, etc.), the date type is made for that.

I wouldn't worry about the data size unless it becomes a problem in the future

Reply:
We shouldn't store a `date` type when we don't have one, else we are back to square one where we decided not to ask for a date for the sake of privacy. Our frontend expects a month number and a year to generate a date in the format like `March 2022` because that's exactly what we asked. Defaulting all the questions asked in the same month to `1` just so that we can store it as `date` seems hacky and adds extra work.

Can you explain more about the `grouping` and `plotting` part?

Reply:
Potential use cases:

1. Sort the questions by the date.
With month and year stored separately, we'll need to do something like ```SORTED BY month, year``` - which I guess is okay, but why not just store it in a way where we can do ```SORTED BY asked_date```?

Now that I think about it, it might just be convenient to go back to the original method we had. We can just store the date as the "best guess date", and it'll be convenient to sort the questions with that date.

2. Use the date field and a points field (we can add it later) as a way to rank the questions - kind of like how Reddit does it.

It'll be nice to be able to use the recency of the question as part of the ranking algorithm in that case.

3. Sort the questions by the date for a specific company.

Similar to 1, but this might be useful for seeing the trend of if/how the questions have changed over time for a specific company.

The more I think about all of this, though, the more it actually makes sense to store the "best guess date" like I was trying to do in the original method.

If 99.9+% of the people just select "within one week", then we can just drop this field and use the current date as the only date we store in the database.

If not, though - wouldn't it be convenient to have that info?

Reply:
If `date` type can be useful I am okay with defaulting to the 1st of the selected month. We can call it `computed_asked_date`  and it will be generated using:
```js
new Date(asked_year, asked_month, 1);
```

Reply:
Coming back to this again - what's the issue with having a "best_guess_date" as we had earlier?

Reply:
I don't recall we had a field with that name. `computed_asked_date` shows that it's related to the asked date and is computed by us however `best_guess_date` conveys no such information.

Reply:
I don't remember what I called it exactly in my original PR, but something like ```estimated_asked_date``` would work.

Reply:
Sounds good. Added the change in dbe71e4

Reply:
Can we just keep the original method I came up with (asking about the time passed since the interview rather than the month)?

It'd be a more accurate approximation of the actual date with that method.

Related: https://github.com/ykdojo/defaang/pull/164#discussion_r967957818